### DISCUSSION TITLE
Default config initialization

### DISCUSSION THREAD

**Discussion by Whaliin** [roflmuffin/CounterStrikeSharp#480] [Ideas]
Instead of having to use field initializers for default values for a config it would be nice to override a method that creates the desired default values for fields.

For example:
You have a field inside your class that is optional to include in the config. 
In the current implementation if this field is not initialized the value in the default config will be empty, and the user making changes to the default configuration will not necessarily understand how to format it to make it work as intended. 

If you define the default value using a field initialization but the users config file is now missing this "optional" field, the config that loads will use the initialized value from the field initialization which is not always what you want.

Example:
![image](https://github.com/roflmuffin/CounterStrikeSharp/assets/1338057/f2321757-5c60-4520-acea-7b794ef9f35e)

If the users config is missing this field, the field will be initialized like above. To work around this, they **have** to define `MapConfigs` in their config file but make it empty so that the deserializer will override the default instance of this field.

**Comment by KillStr3aK** [comment]
not to be nitpicker (just to clarify), these are properties and not fields.
if this is about automatically updating the configuration file when a new property is added to the configuration class, thats in the todo list already

when you define a new property, you should initialize it with a default value otherwise it will be null in case of nullable types, and not optional
